src/fragments/lib/restapi/js/authz.mdx

```diff
@@ -1,96 +1,122 @@
-When working with a REST endpoint, you may need to set request headers for authorization purposes. This is done by passing a `custom_header` function into the configuration:
+When determining the authorization mode for your REST endpoint, there are a few customizations you can do.
+
+## IAM Authorization
+
+By default, the API will be using IAM authorization and the requests will be signed for you automatically. IAM authorization has two modes: one using an **unauthenticated** role, and one using an **authenticated** role. When the user has not signed in, the unauthenticated role is used by default. Once the user has signed in, the authenticate role is used, instead.
+
+When you created your REST API with the Amplify CLI, you were asked if you wanted to restrict access. If you selected **no**, then the unauthenticated role will have access to the API. If you selected **yes**, you would have configured more fine grain access to your API.
+
+## API Key
+
+If you want to configure a public REST API, you can set an API key in Amazon API Gateway. Then, you can set the API key header in the API category configuration. The API key header will be applied to all requests.
 
 ```javascript
-Amplify.configure({
+Amplify.configure(awsconfig, {
   API: {
-    endpoints: [
-      {
-        name: 'sampleCloudApi',
-        endpoint: 'https://xyz.execute-api.us-east-1.amazonaws.com/Development',
-        custom_header: async () => {
-          return { Authorization: 'token' };
-          // Alternatively, with Cognito User Pools use this:
-          // return { Authorization: `Bearer ${(await Auth.currentSession()).getAccessToken().getJwtToken()}` }
-          // return { Authorization: `Bearer ${(await Auth.currentSession()).getIdToken().getJwtToken()}` }
-        }
+    REST: {
+      headers: async () => {
```

Comment:
Right now we don't have a easy way to retrieve the current options when specifying addition options. I will create a dev task for it.